libs/atomvmlib/.gitignore

```diff
@@ -0,0 +1,6 @@
+# Copyright 2023 Fred Dushin <fred@dushin.net>
+#
+# SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+rebar.config
+rebar.lock
+_checkouts/*
```

Comment:
Sources directory is getting polluted in different places:
```
        libs/atomvmlib/_checkouts/
        libs/atomvmlib/rebar.config
        libs/atomvmlib/rebar.lock
        libs/eavmlib/rebar.config
        libs/eavmlib/rebar.lock
        libs/esp32boot/_checkouts/
        libs/esp32boot/rebar.config
        libs/esp32boot/rebar.lock
        libs/esp32devmode/_checkouts/
        libs/esp32devmode/rebar.config
        libs/esp32devmode/rebar.lock
        libs/estdlib/rebar.config
        libs/estdlib/rebar.lock
        libs/etest/rebar.config
        libs/etest/rebar.lock
        tests/code_load/rebar.config
        tests/code_load/rebar.lock
        tests/erlang_tests/include/code_load_pack_data.hrl
        tests/erlang_tests/include/export_test_module_data.hrl
        tests/erlang_tests/rebar.config
        tests/erlang_tests/rebar.lock
        tests/libs/eavmlib/_checkouts/
        tests/libs/eavmlib/rebar.config
        tests/libs/eavmlib/rebar.lock
        tests/libs/estdlib/_checkouts/
        tests/libs/estdlib/rebar.config
        tests/libs/estdlib/rebar.lock
```

We need to fix this before merging. Adding `_checkouts/*` to .gitignore is a workaround.

Reply:
In progress...